internal/nginx/config/template.go

```diff
@@ -7,7 +7,20 @@ import (
 )
 
 var httpServersTemplate = `{{ range $s := .Servers }}
+{{ if $s.IsDefault }}
```

Comment:
I removed the default server block from the nginx-gateway manifest and added logic to generate a default 80 server when there's an HTTP listener with attached routes. 

This may have to be changed in a subsequent PR if we get input that a 404/reject handshake should be returned in the case where a listener is configured with no attached routes.